### ISSUE TITLE
PubSub emulator not triggering onMessagePublished

### ISSUE THREAD

**Issue by Zelfapp** [firebase/firebase-tools#6675]
### [REQUIRED] Environment info

**firebase-tools:** 13.0.2

**Platform:** Ubuntu 22.04

### index.ts
```node
exports.testPubSub = require('./src/testPubSub')
```

### /src/textPubSub.ts
```node
import { onMessagePublished } from 'firebase-functions/v2/pubsub'
import { Encodings, PubSub, Schema, SchemaTypes } from '@google-cloud/pubsub'
import { logger } from 'firebase-functions/v2'
import { onRequest } from 'firebase-functions/v2/https'
import express from 'express'
import cors from 'cors'
import { get } from 'lodash'
const kDebug = process.env.FUNCTIONS_EMULATOR === 'true'
const kMyTopic = 'my-topic'
const app = express()

// Automatically allow cross-origin requests
app.use(cors({ origin: true }))
// Support json encoded bodies and url encoded bodies
app.use(express.urlencoded({ extended: true }))

/**
 * Retrieves a PubSub schema and checks if the provided schema exists.
 *
 * @param {string} schemaName The schema name to get.
 * @return {Promise<boolean>} A promise that resolves to a bool of
 * whether the schema exists.
 */
async function hPubSubSchemaExists(schemaName: string): Promise<boolean> {
  try {
    const pubsub = new PubSub()
    for await (const s of pubsub.listSchemas()) {
      const sName = s.name.split('/').at(-1)

      if (kDebug) {
        logger.log(`Schema path ${s.name}`)
        logger.log(`Just schema name ${sName}`)
      }

      if (sName === schemaName) {
        return true
      }
    }

    return false
  } catch (error) {
    if (kDebug) {
      logger.log(error)
    }
    return false
  }
}

/**
 * Retrieves a PubSub schema.
 *
 * @param {string} schemaName The schema name to get.
 * @return {Promise<Schema>} A promise that resolves to a Schema.
 */
async function hGetPubSubSchema(schemaName: string): Promise<Schema> {
  try {
    const pubsub = new PubSub()
    return pubsub.schema(schemaName)
  } catch (error) {
    if (kDebug) {
      logger.log(error)
    }
    return null
  }
}

/**
 * Retrieves an array of PubSub topics and checks if the provided topic exists.
 *
 * @param {string} topicName The topic to filter for.
 * @return {Promise<boolean>} A promise that resolves to an array of topics.
 */
async function hPubSubTopicExists(topicName: string): Promise<boolean> {
  try {
    const pubsub = new PubSub()
    const [topics] = await pubsub.getTopics()
    return !!topics.filter(topic => topic.name.includes(topicName))?.[0]
  } catch (error) {
    if (kDebug) {
      logger.log(error)
    }
    return false
  }
}

/**
 * Publish a message to Pub/Sub topic.
 * @return {boolean} Whether the message was published.
 */
async function hPublishGenerateRunMessage(): Promise<boolean> {
  try {
    const pubSubClient = new PubSub({
      projectId: 'usamm-app',
    })

    if (kDebug) {
      if (await hPubSubTopicExists(kMyTopic)) {
        await pubSubClient.topic(kMyTopic).delete()
        logger.log(`Deleted ${kMyTopic} topic.`)
      }

      if (await hPubSubTopicExists(kMyTopic)) {
        logger.log(`${kMyTopic} topic already exists.`)
      }

      if (!(await hPubSubTopicExists(kMyTopic))) {
        logger.log(`${kMyTopic} topic does not exist, creating.`)

        let schema: Schema

        if (!(await hPubSubSchemaExists(kMyTopic))) {
          // https://github.com/googleapis/nodejs-pubsub/blob/main/samples/createAvroSchema.js
          schema = await pubSubClient.createSchema(
            kMyTopic,
            SchemaTypes.Avro,
            JSON.stringify({
              type: 'record',
              name: 'Avro',
              fields: [
                {
                  name: 'generateRun',
                  type: 'boolean',
                },
              ],
            }),
          )
        } else {
          schema = await hGetPubSubSchema(kMyTopic)
        }

        // https://github.com/googleapis/nodejs-pubsub/blob/main/samples/createTopicWithSchema.js
        await pubSubClient.createTopic({
          name: kMyTopic,
          schemaSettings: {
            schema: await schema.getName(),
            encoding: Encodings.Json,
          },
          messageStoragePolicy: {
            allowedPersistenceRegions: ['us-east4'],
          },
        })
      }
    }

    const [messageId] = await pubSubClient.topic(kMyTopic).publishMessage({
      data: Buffer.from(JSON.stringify({ generateRun: true })),
    })

    if (kDebug) {
      logger.log(`messageId: ${messageId} published to ${kMyTopic}.`)
    }

    return true
  } catch (error) {
    logger.error(error)
    return false
  }
}

/**
 * Init publishing a message to 'my-topic'
 */
app.get('/init', async (req, res) => {
  try {
    if (!kDebug) {
      throw Error('Up to no good.')
    }

    await hPublishGenerateRunMessage()
    res.send({ success: true })
  } catch (error) {
    res.status(400).send({ success: false, message: error.toString() })
  }
})

exports.endpoint = onRequest({ timeoutSeconds: 1800, memory: '16GiB', cpu: 4 }, app)

/**
 * When a message is published to the generate-packing-slip-run topic, this
 * will be triggered, and generate a packing slip run.
 */
exports.generateRun = onMessagePublished(
  {
    topic: kMyTopic,
    timeoutSeconds: 1800,
    memory: '16GiB',
    cpu: 4,
    region: 'us-east4',
    maxInstances: 1,
  },
  async event => {
    const messageId = get(event, 'data.message.messageId', null)

    try {
      if (!messageId) {
        throw Error('Missing messageId')
      }

      if (kDebug) {
        logger.log(`Processing ${kMyTopic} message id ${messageId}`)
      }
    } catch (error) {
      logger.error(error)
    }
  },
)
```

### firebase.json
```json
{
  "functions": [
    {
      "source": "functions",
      "ignore": [
        "node_modules",
        ".git",
        ".cache",
        "firebase-export*",
        ".vscode",
        ".idea",
        "*.log",
        "*.html",
        "README.md",
        "cache"
      ],
      "predeploy": [
        "npm --prefix \"$RESOURCE_DIR\" run lint",
        "npm --prefix \"$RESOURCE_DIR\" run build"
      ],
      "codebase": "default"
    }
  ],
  "firestore": {
    "database": "my-app",
    "rules": "firestore.rules",
    "indexes": "firestore.indexes.json"
  },
  "hosting": {
    "public": "build/web",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ],
    "rewrites": [
      {
        "source": "**",
        "destination": "/index.html"
      }
    ]
  },
  "storage": {
    "rules": "storage.rules"
  },
  "emulators": {
    "auth": {
      "host": "0.0.0.0",
      "port": 9099
    },
    "functions": {
      "host": "0.0.0.0",
      "port": 5001
    },
    "firestore": {
      "host": "0.0.0.0",
      "port": 8088
    },
    "hosting": {
      "host": "0.0.0.0",
      "port": 5003
    },
    "storage": {
      "host": "0.0.0.0",
      "port": 9199
    },
    "pubsub": {
      "host": "0.0.0.0",
      "port": 8085
    },
    "ui": {
      "enabled": true
    },
    "singleProjectMode": true
  }
}
```

### [REQUIRED] Steps to reproduce

1. Call http://127.0.0.1:5001/my-app/us-east4/testPubSub-endpoint/init to publish a message to `my-topic`
3. See if `onMessagePublished` logs anything.

### [REQUIRED] Expected behavior

Expecting to see that the `onMessagePublished` function is triggered when posting to `my-topic`.

### [REQUIRED] Actual behavior

So far, the `onMessagePublished` is never triggered by posting messages to `my-topic`. While I can confirm the messages are published, and I believe locally, I'm clearly missing a configuration or a reference to a doc somewhere to setup PubSub onMessagePublished emulator correctly.

**Comment by joehan** [comment]
Hey @Zelfapp, sorry to hear that you are running into issues here. The first thing I'd like to check is whether messages are being successfully published to the pubsub emulator. Could you share:
- the contents of  your `firebase.json` file. If you don't want to share this, just the `emulators` section should suffice
- the full console output when you replicate the issue with the `--debug` flag included

**Comment by Zelfapp** [comment]
@joehan I just updated the code. It is actually fully reproducible now, rather than snippets of my code. So, should be easy to test with what I'm trying to do. And added `firebase.json`.

Without `--debug` when I call the test endpoint http://127.0.0.1:5001/my-app/us-east4/testPubSub-endpoint/init, I get the following, but it does not have the messageId log in the `onMessagePublished` function. So it is not getting triggered?

```
i  functions: Beginning execution of "us-east4-testPubSub-endpoint"
>  {"severity":"INFO","message":"Deleted my-topic topic."}
>  {"severity":"INFO","message":"my-topic topic does not exist, creating."}
>  {"severity":"INFO","message":"messageId: 1 published to my-topic."}
i  functions: Finished "us-east4-testPubSub-endpoint" in 117.359316ms
```

With `--debug`, when I call my test endpoint that publishes a message to `my-topic`: http://127.0.0.1:5001/my-app/us-east4/testPubSub-endpoint/init, I get the following:

```
[2024-01-09T18:40:44.068Z] [work-queue] {"queuedWork":["/usamm-app/us-east4/testPubSub-endpoint/init-2024-01-09T18:40:44.068Z"],"queueLength":1,"runningWork":[],"workRunningCount":0}
[2024-01-09T18:40:44.069Z] [work-queue] {"queuedWork":[],"queueLength":0,"runningWork":["/usamm-app/us-east4/testPubSub-endpoint/init-2024-01-09T18:40:44.068Z"],"workRunningCount":1}
[2024-01-09T18:40:44.070Z] Accepted request GET /usamm-app/us-east4/testPubSub-endpoint/init --> us-east4-testPubSub-endpoint
[2024-01-09T18:40:44.072Z] [functions] Runtime ready! Sending request! {"metadata":{"emulator":{"name":"functions"},"message":"[functions] Runtime ready! Sending request!"}}
[2024-01-09T18:40:44.073Z] [functions] Got req.url=/usamm-app/us-east4/testPubSub-endpoint/init, mapping to path=/init {"metadata":{"emulator":{"name":"functions"},"message":"[functions] Got req.url=/usamm-app/us-east4/testPubSub-endpoint/init, mapping to path=/init"}}
[2024-01-09T18:40:44.077Z] [worker-pool] addWorker(us-east4-testPubSub-endpoint) {"metadata":{"emulator":{"name":"functions"},"message":"[worker-pool] addWorker(us-east4-testPubSub-endpoint)"}}
[2024-01-09T18:40:44.078Z] [worker-pool] Adding worker with key us-east4-testPubSub-endpoint, total=1 {"metadata":{"emulator":{"name":"functions"},"message":"[worker-pool] Adding worker with key us-east4-testPubSub-endpoint, total=1"}}
[2024-01-09T18:40:44.345Z] [runtime-status] [1304534] Resolved module firebase-admin {"declared":true,"installed":true,"version":"11.11.1","resolution":"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-admin/lib/index.js"} {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] Resolved module firebase-admin {\"declared\":true,\"installed\":true,\"version\":\"11.11.1\",\"resolution\":\"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-admin/lib/index.js\"}"}}
[2024-01-09T18:40:44.345Z] [runtime-status] [1304534] Resolved module firebase-functions {"declared":true,"installed":true,"version":"4.6.0","resolution":"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-functions/lib/v1/index.js"} {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] Resolved module firebase-functions {\"declared\":true,\"installed\":true,\"version\":\"4.6.0\",\"resolution\":\"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-functions/lib/v1/index.js\"}"}}
[2024-01-09T18:40:44.345Z] [runtime-status] [1304534] Outgoing network have been stubbed. [{"name":"http","status":"mocked"},{"name":"http","status":"mocked"},{"name":"https","status":"mocked"},{"name":"https","status":"mocked"},{"name":"net","status":"mocked"}] {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] Outgoing network have been stubbed. [{\"name\":\"http\",\"status\":\"mocked\"},{\"name\":\"http\",\"status\":\"mocked\"},{\"name\":\"https\",\"status\":\"mocked\"},{\"name\":\"https\",\"status\":\"mocked\"},{\"name\":\"net\",\"status\":\"mocked\"}]"}}
[2024-01-09T18:40:44.346Z] [runtime-status] [1304534] Resolved module firebase-functions {"declared":true,"installed":true,"version":"4.6.0","resolution":"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-functions/lib/v1/index.js"} {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] Resolved module firebase-functions {\"declared\":true,\"installed\":true,\"version\":\"4.6.0\",\"resolution\":\"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-functions/lib/v1/index.js\"}"}}
[2024-01-09T18:40:44.399Z] [runtime-status] [1304534] Checked functions.config() {"config":{}} {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] Checked functions.config() {\"config\":{}}"}}
[2024-01-09T18:40:44.399Z] [runtime-status] [1304534] firebase-functions has been stubbed. {"functionsResolution":{"declared":true,"installed":true,"version":"4.6.0","resolution":"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-functions/lib/v1/index.js"}} {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] firebase-functions has been stubbed. {\"functionsResolution\":{\"declared\":true,\"installed\":true,\"version\":\"4.6.0\",\"resolution\":\"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-functions/lib/v1/index.js\"}}"}}
[2024-01-09T18:40:44.399Z] [runtime-status] [1304534] Resolved module firebase-functions {"declared":true,"installed":true,"version":"4.6.0","resolution":"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-functions/lib/v1/index.js"} {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] Resolved module firebase-functions {\"declared\":true,\"installed\":true,\"version\":\"4.6.0\",\"resolution\":\"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-functions/lib/v1/index.js\"}"}}
[2024-01-09T18:40:44.400Z] [runtime-status] [1304534] Resolved module firebase-admin {"declared":true,"installed":true,"version":"11.11.1","resolution":"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-admin/lib/index.js"} {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] Resolved module firebase-admin {\"declared\":true,\"installed\":true,\"version\":\"11.11.1\",\"resolution\":\"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-admin/lib/index.js\"}"}}
[2024-01-09T18:40:44.400Z] [runtime-status] [1304534] Resolved module firebase-functions {"declared":true,"installed":true,"version":"4.6.0","resolution":"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-functions/lib/v1/index.js"} {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] Resolved module firebase-functions {\"declared\":true,\"installed\":true,\"version\":\"4.6.0\",\"resolution\":\"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-functions/lib/v1/index.js\"}"}}
[2024-01-09T18:40:44.401Z] [runtime-status] [1304534] firebase-admin has been stubbed. {"adminResolution":{"declared":true,"installed":true,"version":"11.11.1","resolution":"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-admin/lib/index.js"}} {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] firebase-admin has been stubbed. {\"adminResolution\":{\"declared\":true,\"installed\":true,\"version\":\"11.11.1\",\"resolution\":\"/home/nate/AndroidStudioProjects/order_fulfillment/functions/node_modules/firebase-admin/lib/index.js\"}}"}}
[2024-01-09T18:40:44.401Z] [runtime-status] [1304534] initializeApp(DEFAULT) {"storageBucket":"usamm-app.appspot.com","databaseURL":"https://usamm-app-default-rtdb.firebaseio.com","projectId":"usamm-app"} {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] initializeApp(DEFAULT) {\"storageBucket\":\"usamm-app.appspot.com\",\"databaseURL\":\"https://usamm-app-default-rtdb.firebaseio.com\",\"projectId\":\"usamm-app\"}"}}
[2024-01-09T18:40:44.976Z] [runtime-status] [1304534] Functions runtime initialized. {"cwd":"/home/nate/AndroidStudioProjects/order_fulfillment/functions","node_version":"18.19.0"} {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] Functions runtime initialized. {\"cwd\":\"/home/nate/AndroidStudioProjects/order_fulfillment/functions\",\"node_version\":\"18.19.0\"}"}}
[2024-01-09T18:40:44.977Z] [runtime-status] [1304534] Listening to port: /tmp/fire_emu_29c8503edf9a99e8.sock {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[runtime-status] [1304534] Listening to port: /tmp/fire_emu_29c8503edf9a99e8.sock"}}
[2024-01-09T18:40:44.989Z] [worker-us-east4-testPubSub-endpoint-ca33a002-db89-4f46-9780-b1ee0067df7f]: IDLE {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[worker-us-east4-testPubSub-endpoint-ca33a002-db89-4f46-9780-b1ee0067df7f]: IDLE"}}
[2024-01-09T18:40:44.989Z] [worker-pool] submitRequest(triggerId=us-east4-testPubSub-endpoint) {"metadata":{"emulator":{"name":"functions"},"message":"[worker-pool] submitRequest(triggerId=us-east4-testPubSub-endpoint)"}}
i  functions: Beginning execution of "us-east4-testPubSub-endpoint" {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"Beginning execution of \"us-east4-testPubSub-endpoint\""}}
[2024-01-09T18:40:44.990Z] [worker-us-east4-testPubSub-endpoint-ca33a002-db89-4f46-9780-b1ee0067df7f]: BUSY {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[worker-us-east4-testPubSub-endpoint-ca33a002-db89-4f46-9780-b1ee0067df7f]: BUSY"}}
[2024-01-09T18:40:45.030Z] Jan 09, 2024 11:40:45 AM io.gapi.emulators.netty.HttpVersionRoutingHandler channelRead
INFO: Detected HTTP/2 connection.
 {"metadata":{"emulator":{"name":"pubsub"},"message":"Jan 09, 2024 11:40:45 AM io.gapi.emulators.netty.HttpVersionRoutingHandler channelRead\nINFO: Detected HTTP/2 connection.\n"}}
>  {"severity":"INFO","message":"Deleted my-topic topic."} {"user":{"severity":"INFO","message":"Deleted my-topic topic."},"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"\u001b[90m> \u001b[39m {\"severity\":\"INFO\",\"message\":\"Deleted my-topic topic.\"}"}}
>  {"severity":"INFO","message":"my-topic topic does not exist, creating."} {"user":{"severity":"INFO","message":"my-topic topic does not exist, creating."},"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"\u001b[90m> \u001b[39m {\"severity\":\"INFO\",\"message\":\"my-topic topic does not exist, creating.\"}"}}
>  {"severity":"INFO","message":"messageId: 1 published to my-topic."} {"user":{"severity":"INFO","message":"messageId: 1 published to my-topic."},"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"\u001b[90m> \u001b[39m {\"severity\":\"INFO\",\"message\":\"messageId: 1 published to my-topic.\"}"}}
[2024-01-09T18:40:45.116Z] Finishing up request with event=pause {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"Finishing up request with event=pause"}}
i  functions: Finished "us-east4-testPubSub-endpoint" in 126.088407ms {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"Finished \"us-east4-testPubSub-endpoint\" in 126.088407ms"}}
[2024-01-09T18:40:45.116Z] [worker-us-east4-testPubSub-endpoint-ca33a002-db89-4f46-9780-b1ee0067df7f]: IDLE {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"[worker-us-east4-testPubSub-endpoint-ca33a002-db89-4f46-9780-b1ee0067df7f]: IDLE"}}
[2024-01-09T18:40:45.116Z] Finishing up request with event=finish {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"Finishing up request with event=finish"}}
[2024-01-09T18:40:45.116Z] Finishing up request with event=close {"metadata":{"emulator":{"name":"functions"},"function":{"name":"us-east4-testPubSub-endpoint"},"extension":{},"message":"Finishing up request with event=close"}}
[2024-01-09T18:40:45.116Z] [work-queue] {"queuedWork":[],"queueLength":0,"runningWork":[],"workRunningCount":0}
```

**Comment by Zelfapp** [comment]
@joehan do you see any logic errors or is this a bug in PubSub emulator in 13.0.2 onMessagePublished?

**Comment by Zelfapp** [comment]
@joehan pinging. Or Firebase in general. Is this an issue with firebase-tools or a logic error?

**Comment by aalej** [comment]
Hey @Zelfapp, apologies for the delay. I tried to repro this, it took a while to replicate and narrow down the cause of the issue. From what I can gather, the issue starts to occur after deleting the Pub/Sub topic. After removing the code which deletes the topic, the `onMessagePublished` function is triggered as expected.

These are the lines I commented out:
```ts
if (await hPubSubTopicExists(kMyTopic)) {
  await pubSubClient.topic(kMyTopic).delete()
  logger.log(`Deleted ${kMyTopic} topic.`)
}
```

Logs when calling the endpoint "http://127.0.0.1:5001/<project_id>/us-central1/endpoint/init"
```
i  functions: Beginning execution of "us-central1-endpoint"
>  {"severity":"INFO","message":"my-topic topic already exists."}
>  {"severity":"INFO","message":"messageId: 1 published to my-topic."}
i  functions: Finished "us-central1-endpoint" in 184.705609ms
i  functions: Beginning execution of "us-east4-generateRun"
>  {"severity":"INFO","message":"Processing my-topic message id 1"}
i  functions: Finished "us-east4-generateRun" in 15.880214ms
```

We're currently verifying whether this is a bug or a working as intended behavior

**Comment by aalej** [comment]
Hey @Zelfapp, just a quick update here. I discussed this with our team, and it seems like it’s expected that deleting a Pub/Sub topic will break the trigger since it's the Pub/Sub subscription that's relevant in function triggering, not just matching Pub/Sub topic name. I also tried deploying this on an actual project and got the same results. 

I would recommend to avoid deleting and re-creating the Pub/Sub topic as this could cause unexpected behaviors.

For now I’m going to close this since this looks like it’s working as intended, but feel free to ask this to be reopened if you still encounter issues with triggering the `onMessagePublished` functions.
